Makefile

```diff
@@ -55,8 +65,18 @@ vet: ## Run go vet against code.
 	go vet ./...
 
 .PHONY: lint
-lint: ## Run golangci-lint against code.
-	golangci-lint run ./...
+lint: ## Run golint against code.
+	golint ./...
+
+.PHONY: vendor/check
+vendor/check: vendor/fix
+	git diff --exit-code vendor/
+	git diff --exit-code go.sum
+
+.PHONY: vendor/fix
+vendor/fix:
+	go mod tidy
+	go mod vendor
```

Comment:
We haven;t vendored module sup to now, do we really want to?

Reply:
We would eventually, right? 🤔

Reply:
I have no strong opinions on whether we vendor now or not, pros and cons to it. Eventually I'd imagine we will be asked to, but it's not IMO essential right now, and we are only really hitting cons (messy commit history with vendor changes).

Point here is though, we aren't vendoring, and you are adding a check to check the vendor directory. So either this PR needs to add the vendor directory as well, and we vendor from now on, or we shoudl just remove the vendor related targets.

@maleck13 @philbrookes Any opinions on vendoring?

Reply:
lots of opinions on vendoring :) but I agree not something we need to do at this point

Reply:
then I'll go with a simple formating check and will drop vendors :thinking: